Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -3152,40 +3152,16 @@ static String commandNameForSelector(SEL selector)
     m_page->capitalizeWord();
 }
 
-#if USE(APPLE_INTERNAL_SDK)
-#include <WebKitAdditions/WebViewImplAdditions.mm>
-#else
-static Vector<WebCore::CompositionUnderline> extractInitialUnderlines(NSAttributedString *)
-{
-    return { };
-}
-
-bool WebViewImpl::markedTextInputEnabled() const
-{
-    return false;
-}
-
-void WebViewImpl::showMarkedTextForCandidate(NSTextCheckingResult *, NSRange, NSRange)
-{
-}
-
-void WebViewImpl::showMarkedTextForCandidates(NSArray<NSTextCheckingResult *> *)
-{
-}
-
 NSTextCheckingTypes WebViewImpl::getTextCheckingTypes() const
 {
-    return NSTextCheckingTypeSpelling | NSTextCheckingTypeReplacement | NSTextCheckingTypeCorrection;
-}
+    NSTextCheckingTypes types = NSTextCheckingTypeSpelling | NSTextCheckingTypeReplacement | NSTextCheckingTypeCorrection;
+#if HAVE(INLINE_PREDICTIONS)
+    if (allowsInlinePredictions())
+        types |= (NSTextCheckingType)_NSTextCheckingTypeSingleCompletion;
```

Comment:
Is there an API value for `_NSTextCheckingTypeSingleCompletion`? If not, this needs an SPI declaration.